pypeit/core/arc.py

```diff
@@ -906,7 +906,9 @@ def detect_lines(censpec, sigdetect=5.0, fwhm=4.0, fit_frac_fwhm=1.25, input_thr
     xrng = np.arange(detns.size, dtype=np.float)
 
     if cont_subtract:
-        cont_now, cont_mask = iter_continuum(detns, inmask=None if bpm is None else np.invert(bpm),
+        # FW: for trimmed orders (i.e. set spec_max or spec_min in spectrograph file), you will have lot zeros.
+        # so I mask those zeros if bpm is not given.
+        cont_now, cont_mask = iter_continuum(detns, inmask= (detns!=0) if bpm is None else np.invert(bpm),
```

Comment:
Seems reasonable to me, but @jhennawi will need to comment on whether this precipitates running the full DevSuite.

Reply:
I would prefer it if you just used the bpm option here in the routine that calls detect_lines. This is a bug that in principle impacts many spectrographs and as such, I think that necessitates a more transparent implementation. I think we never had a problem because it just set the continuum to zero, but we need to be a bit more careful here.

Reply:
@jhennawi  Ok. I just reversed it. It seems this is no longer a problem if I do not use the holy grail method for wavelength. Since I have already archived the wavelength solution. I suppose we do not need to change the inmask anymore.